src/util/simplify_expr.cpp

```diff
@@ -2058,26 +2052,33 @@ bool simplify_exprt::simplify_byte_update(byte_update_exprt &expr)
 
     for(const auto &component : components)
     {
-      mp_integer m_offset=
-        member_offset(struct_type, component.get_name(), ns);
-      mp_integer m_size_bits=pointer_offset_bits(component.type(), ns);
-      mp_integer m_size_bytes=m_size_bits/8;
+      auto m_offset = member_offset(struct_type, component.get_name(), ns);
+
+      auto m_size_bits = pointer_offset_bits(component.type(), ns);
+
+      if(!m_offset.has_value() || !m_size_bits.has_value() || *m_size_bits == 0)
+      {
+        result_expr.make_nil();
+        break;
+      }
 
       // can we determine the current offset, and is it a byte-sized
```

Comment:
Comment belongs above, since the "can we determine" and the byte alignment question are now checked seperately

Reply:
I think this is still wrong -- "Can we determine the current offset?" belongs above `if(!m_offset.has_value() || ...)`, then "Is it a byte-sized member?" belongs here.